test/integration/smoke/test_kubernetes_clusters.py

```diff
@@ -622,11 +704,10 @@ def createKubernetesCluster(self, name, version_id, size=1, control_nodes=1):
         createKubernetesClusterCmd.noderootdisksize = 10
         createKubernetesClusterCmd.account = self.account.name
         createKubernetesClusterCmd.domainid = self.domain.id
+        createKubernetesClusterCmd.clustertype = cluster_type
         if self.default_network:
             createKubernetesClusterCmd.networkid = self.default_network.id
         clusterResponse = self.apiclient.createKubernetesCluster(createKubernetesClusterCmd)
-        if not clusterResponse:
-            self.cleanup.append(clusterResponse)
```

Comment:
if we remove this here , we need to add it in the calling method(s). Alternative to muy suggestion (https://github.com/apache/cloudstack/pull/7515/files#r1217636578) we can validate the response here and on success add the cluster to self.cleanup here.

Reply:
KubernetesCluster class is not implemented and the cleanup call on the response object would fail.

Reply:
I feel this code was unnecessary. Adding clusterResponse in cleanup won't work as we do not have code in marvin/lib/base to handle obj.delete() for a KubernetesCluster.
Also, as far as I remember this test file tries to use the same k8s cluster across different test cases to preserve time to deploy a new k8s cluster so either we refactor the whole class or handle cleanup of created cluster depending on the testcase

Reply:
ok, makes sense, but 
@shwstppr the cluster is created here and so it should be destroyed here, or its creation should be moved to class-scope or higher.
@vishesh92 I think it might be worth adding a base.lib entry for clusters, but ... as said, this is a bit of scope creep. It might however explain the spurious failures on k8s we see in tests runs.

Reply:
as discussed with @vishesh92 ofline, he created https://github.com/apache/cloudstack/issues/7604 to address this. cc @shwstppr